WIP: documentation

campagnola opened the pull request:
I have started with some outlines of what I would like to include in the documentation.
See `intro.rst`, `nodes.rst`, `manager.rst`, `streams.rst`.
Please comment on these!

samuelgarcia commented on doc/apiref/processors.rst:
```diff
@@ -0,0 +1,18 @@
+
+.. autoclass::  pyacq.dsp.trigger.TriggerBase
+   :members:
+
```
TriggerBase is internal maybe not necessary to put in docs.

campagnola replied:
Yesh, agreed.

campagnola commented:
Docs are online to help with review process: http://pyacq.readthedocs.org/en/latest/intro.html

samuelgarcia commented:
This is cool.

campagnola commented:
Questions: 

1. What license?
2. Who are the copyright holder(s)?

campagnola commented:
( Is CNRS the copyright holder? )
3. Should CNRS and funding sources be mentioned in documentation / readme ?

Lx37 commented:
Hi Luke,
What a great job !! I look forward to get time to contribute..
1. I'm still waiting for an official validation of the licence :( but we get an oral agreement about a BSD licence 3-clause, so let's go for it.
2. The copyright holder would be the French National Center for Scientific Research  (CNRS) 
3. I would say yes, CNRS and funding sources should be mentioned in documentation / readme. As founding source we can precise the Lyon Neuroscience Research Center (CRNL).

Here is a proposition for the licence text : Copyright (c) 2016, French National Center for Scientific Research (CNRS)
All rights reserved.

Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions are met:
    * Redistributions of source code must retain the above copyright
      notice, this list of conditions and the following disclaimer.
    * Redistributions in binary form must reproduce the above copyright
      notice, this list of conditions and the following disclaimer in the
      documentation and/or other materials provided with the distribution.
    * Neither the name of the CNRS nor the
      names of its contributors may be used to endorse or promote products
      derived from this software without specific prior written permission.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE HOLDERS AND CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

How could I help you to add it on the code ?
*** Happy New Year *** :)
best

campagnola commented:
License and copyright are updated throughout. I have added a standard header to all py files.

> How could I help you to add it on the code ?

Ah, I apologize, I only read this line after finishing :/

campagnola commented:
Status: close to finished. I'd like to add a few more examples and polish up some rough spots in the API.

Lx37 commented:
The doc is really nice and easy to read and understand :+1:  
Just one note and one question :
- in the introduction example, using a BrainAmp device with a Spikesorter worker is (for me) ambiguous. BrainAmp is, as far as I know, used for EEG but not for intra-recordings. An 'Open Ephys' device or the 'MeasurementComputing' device would be more suitable for the spike sorting worker. But it's a EEG purist note.
- in the same example, why the viewer is independent ? // Is there an advantage to not associating the viewer to the manager ?

I can add examples using EEG devices like the one used for pyaudio ?

campagnola commented:
Ah! You're right, good catch. At some point I switched from an MEA to EEG because I thought the latter would be recognizable to a larger audience, but I didn't consider the consequences.

I've changed it to a DAQmx device for now because I suspect that will be more recognizable than Measurement Computing (OpenEphys would be nice too, but we don't support it yet?)  Do you have a sense of which devices are more commonly used?

I've also corrected the example to create all nodes via the manager.

samuelgarcia commented:
I don't anderstand why they are all that changes on files.

Can I merge this ?
Does the http://pyacq.readthedocs.io point to the master or only this branch ?

campagnola commented:
Most files just had a header added or docstring changes.
Give me a little more time to test; it should be ready soon.

campagnola commented:
Ok, unit tests still pass for me here, and the docs are building. Merge?

campagnola commented:
pyacq.readthedocs is pointed at this branch because master doesn't have a sphinx setup at all. I will redirect it once this is merged, and then see if I can figure out how to give you access to the account..

samuelgarcia commented:
OK I merge this. Thank you very much.
